client/src/components/TypeEditor.vue

```diff
@@ -0,0 +1,227 @@
+<script lang="ts">
+import {
+  defineComponent, reactive, toRef, watch,
+} from '@vue/composition-api';
+import { usePrompt } from 'dive-common/vue-utilities/prompt-service';
+import { useHandler, useTypeStyling, useUsedTypes } from 'vue-media-annotator/provides';
+
+export default defineComponent({
+  name: 'TypeEditor',
+
+  props: {
+    selectedType: {
+      type: String,
+      required: true,
+    },
+  },
+
+  setup(props, { emit }) {
+    const typeStylingRef = useTypeStyling();
+    const usedTypesRef = useUsedTypes();
+    const { prompt } = usePrompt();
+    const {
+      updateTypeName,
+      updateTypeStyle,
+      deleteType,
+    } = useHandler();
+
+    const data = reactive({
+      selectedColor: '',
+      selectedType: '',
+      editingType: '',
+      editingColor: '',
+      editingThickness: 5,
+      editingFill: false,
+      editingOpacity: 1.0,
+      valid: true,
+    });
+
+    function acceptChanges() {
+      if (data.editingType !== data.selectedType) {
+        updateTypeName({
+          currentType: data.selectedType,
+          newType: data.editingType,
+        });
+      }
+      updateTypeStyle({
+        type: data.editingType,
+        value: {
+          color: data.editingColor,
+          strokeWidth: data.editingThickness,
+          fill: data.editingFill,
+          opacity: data.editingOpacity,
+        },
+      });
+      emit('close');
+    }
+
+    async function clickDeleteType(type: string) {
+      const text = `Do you want to delete this empty Type: ${type}`;
+      const result = await prompt({
+        title: 'Confirm',
+        text,
+        confirm: true,
+      });
+      if (result) {
+        deleteType(type);
+        emit('close');
+      }
+    }
+
+    function init() {
+      data.selectedType = props.selectedType;
+      data.editingType = props.selectedType;
+      data.editingColor = typeStylingRef.value.color(props.selectedType);
+      data.editingThickness = typeStylingRef.value.strokeWidth(props.selectedType);
+      data.editingFill = typeStylingRef.value.fill(props.selectedType);
+      data.editingOpacity = typeStylingRef.value.opacity(props.selectedType);
+    }
+    watch(toRef(props, 'selectedType'), init);
+    init();
```

Comment:
`watchEffect` still isnt' good at figuring out what proerties are read vs write.  These are all write only, but they trigger the effect function, so falling back to the old pattern...